python/lsst/meas/algorithms/detection.py

```diff
@@ -489,6 +515,38 @@ def setEdgeBits(maskedImage, goodBBox, edgeBitmask):
                                                      afwGeom.ExtentI(w, h)), afwImage.LOCAL)
             edgeMask |= edgeBitmask
 
+    @pipeBase.timeMethod
+    def replacePeaks(self, targetFpList, goodPeakFpList):
+        """Replace all of the peaks in one FootprintSet with the overlapping ones from another.
+
+        When this would remove all of the Peaks from a Footprint, we instead remove all but the
+        brightest.
+
+        Parameters
+        ----------
+        targetFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to replace.
+        goodPeakFpSet : `list`
+            Sequence containing the Footprints whose Peaks we want to use instead.
+        """
+        for targetFp in targetFpList:
+            targetBBox = targetFp.getBBox()
+            newPeaks = afwDet.PeakCatalog(targetFp.getPeaks().getTable())
+            for goodPeakFp in goodPeakFpList:
```

Comment:
This double `for` is going to chew up the runtime.  Can we do it a bit more cleverly?  Maybe remove items from the `goodPeakFpList` when they get copied over?  Can you order the footprints so that you can cease iterating (no need to even check the bbox overlap) once you get past a certain point?